app/services/optout/OptOutService.scala

```diff
@@ -97,74 +89,97 @@ class OptOutService @Inject()(itsaStatusUpdateConnector: ITSAStatusUpdateConnect
-                             (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[OptOutUpdateResponse] = {
-
-    setupOptOutProposition().flatMap { proposition =>
-      if (proposition.isOneYearOptOut)
-        makeOptOutUpdateRequestForOneYear(proposition)
-
-      else {
-        taxPayerIntent.flatMap(intent => proposition.optOutTaxYearFor(intent))
-          .map(optOutTaxYearIntent => makeOptOutUpdateRequest(proposition, optOutTaxYearIntent))
-          .getOrElse(Future.successful(OptOutUpdateResponseFailure.defaultFailure()))
-      }
+  def makeOptOutUpdateRequest()(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[OptOutUpdateResponse] = {
+    fetchOptOutProposition().flatMap { proposition =>
+      proposition.optOutPropositionType.map {
+        case _: OneYearOptOutProposition =>
+          saveIntent(intent = proposition.availableTaxYearsForOptOut.head)
+          makeOptOutUpdateRequest(proposition)
+        case _: MultiYearOptOutProposition => makeOptOutUpdateRequest(proposition)
+      } getOrElse Future.successful(OptOutUpdateResponseFailure.defaultFailure())
     }
   }
 
+  def saveIntent(intent: TaxYear)(implicit hc: HeaderCarrier): Future[Boolean] = {
+    val data = UIJourneySessionData(
+      sessionId = hc.sessionId.get.value,
+      journeyType = OptOutJourney.Name,
+      optOutSessionData = Some(OptOutSessionData(selectedOptOutYear = Some(intent.toString)))
+    )
+    repository.set(data)
+  }
 
-  def makeOptOutUpdateRequest(optOutProposition: OptOutProposition, intent: OptOutTaxYear)
+  def makeOptOutUpdateRequest(optOutProposition: OptOutProposition)
                              (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[OptOutUpdateResponse] = {
 
-    val optOutYearsToUpdate = optOutProposition.optOutYearsToUpdate(intent)
+    def makeUpdateCalls(optOutYearsToUpdate: Seq[OptOutTaxYear]): Seq[Future[OptOutUpdateResponse]] = {
+      optOutYearsToUpdate.map(optOutYear => itsaStatusUpdateConnector.requestOptOutForTaxYear(optOutYear.taxYear, user.nino, optOutUpdateReason))
+    }
+
+    def findAnyFailOrFirstSuccess(responses: Seq[OptOutUpdateResponse]): OptOutUpdateResponse = {
+      responses.find(_.isInstanceOf[OptOutUpdateResponseFailure]).getOrElse(responses.head)
+    }
 
-    val responses: Seq[Future[OptOutUpdateResponse]] = optOutYearsToUpdate.map(optOutYear =>
-      itsaStatusUpdateConnector.requestOptOutForTaxYear(optOutYear.taxYear, user.nino, optOutUpdateReason))
+    val result = for {
+      intentTaxYear <- OptionT(fetchSavedIntent())
+      intentOptOutTaxYear <- OptionT(Future.successful(optOutProposition.optOutTaxYearFor(intentTaxYear)))
+      yearsToUpdate <- OptionT(Future.successful(Option(optOutProposition.optOutYearsToUpdate(intentOptOutTaxYear))))
+      responsesSeqOfFutures  <- OptionT(Future.successful(Option(makeUpdateCalls(yearsToUpdate))))
```

Comment:
should fix: use Some eg. Option(makeUpdateCalls(yearsToUpdate)) -> Some(makeUpdateCalls(yearsToUpdate)) to represent value. 

update similar in following code

Reply:
done